server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -458,6 +460,27 @@ private void write(final Collection<CommitSession> sessions, boolean mincFinish,
       // the logs haven't changed.
       final int finalCurrent = currentLogId;
       if (!success) {
+        final ServiceLock tabletServerLock = tserver.getLock();
+        if (sawWriteFailure && tabletServerLock != null) {
```

Comment:
A previous iteration of this was also halting if tabletServerLock was null.  Why did that change?

Reply:
Looks like I was trying to prevent a NPE and ended up changing the logic. I will fix.

Reply:
Changed in 5b95b50